File: algorithms/ar-lcs/python3/ar-lcs.py
def lcs(string1, string2, get_string=False):
    m = len(string1) 
    n = len(string2) 
   
    dp = [[0]*(n+1) for i in range(m+1)] 
  
    for i in range(1, m+1): 
        for j in range(1, n+1): 
            if string1[i-1] == string2[j-1]: 
                dp[i][j] = dp[i-1][j-1] + 1
            else: 
                dp[i][j] = max(dp[i-1][j] , dp[i][j-1]) 

    if not get_string:
        return dp[m][n]

    idx = dp[m][n]
    lcs = [""] * (idx + 1) 
    lcs[idx] = "" 
    
    # Backtrack
    i = m 
    j = n 
    while i > 0 and j > 0: 
        if string1[i-1] == string2[j-1]: 
            lcs[idx-1] = string1[i-1] 
            i -= 1
            j -= 1
            idx -= 1
        elif dp[i-1][j] > dp[i][j-1]: 
            i -= 1
        else: 
            j -= 1
    
    return "".join(lcs)

if __name__ == '__main__':
    print(lcs('AGGTAB', 'GXTXAYB'))
    print(lcs('AGGTAB', 'GXTXAYB', get_string=True))

File: algorithms/ar-lcs/javascript/ar-lcs.js

function lcs(a, b){
    var m = a.length, n = b.length, i, j, l = [];
    for(i = 0; i <= m; i++){
        l[i] = [];
        for(j = 0; j <= n; j++){
            if(i == 0 || j == 0){
                l[i][j] = 0;
            }else if(a[i-1] == b[j-1]){
                l[i][j] = l[i-1][j-1] + 1;
            }else{
                l[i][j] = Math.max(l[i-1][j], l[i][j-1]);
            }
        }
    }
    return l[m][n];
}

let a = [1,2,3,4,5,6,7,8,9,0];
let b = [2,4,6,7,9,10];

console.log(lcs(a, b));